### PR TITLE
[MID-33] set up styled components

### PR DESCRIPTION
**Pull request by Mykhailo-Pasternak**
ticket: https://trello.com/c/dAf1aAfY/33-investigate-ui-components-part

**Brief Description:** 

- Add configuration.
- Add global styles.
- Add a theme (+ add a small example of how to use).

### CODE REVIEW

File: src/app/layout.js [hunk 1]
```diff
@@ -1,23 +1,24 @@
-import { Inter } from "next/font/google";
+'use client';
```
Comment (alex-popov-tech) [inline src/app/layout.js:1]: 🔴  isn't using 'use client' here hucks up SSR for whole project? 🤔

File: src/app/Providers.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+'use client';
+import StyledComponentsRegistry from '../lib/registry'
+import { ThemeProvider } from 'styled-components';
+import theme from './styles/theme';
+
+const Providers = (props) => {
```
Comment (alex-popov-tech) [inline src/app/Providers.js]: 🔴  I would prefer a bit more descriptive name than just 'Provider' here

File: src/app/page.js [hunk 3]
```diff
@@ -1,3 +1,13 @@
+'use client';
```
Comment (alex-popov-tech) [inline src/app/page.js]: 🔴  again, using 'use client' disables ssr on top level for that page, is that ok?

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (alex-popov-tech) [inline tsconfig.json]: why do you need tsconfig in js project?

File: src/app/page.js (hunk 3 above)
Comment (Mykhailo-Pasternak) [reply src/app/page.js]: image

look at the source. Global styles and styles are already arriving.

Comment (Mykhailo-Pasternak) [reply src/app/page.js]: I checked if SSR works like this. If I'm wrong, let me know.

File: tsconfig.json (hunk 4 above)
Comment (Mykhailo-Pasternak) [reply tsconfig.json]: accidentally created.
Deleted it.

File: src/app/page.js (hunk 3 above)
Comment (watchlar2000) [reply src/app/page.js]: You can avoid using "use client" here by moving MyDiv styled component into "_components" folder. Then you'll be able to just import "MyDiv" whenever you want, including server side "Home" component. Give it a try.

Comment (Mykhailo-Pasternak) [reply src/app/page.js]: page hydration happens on the server side, and returns HTML with styles to the client.

Comment (Mykhailo-Pasternak) [reply src/app/page.js]: @watchlar2000 I know it. This is just a small example for demonstration.

**Review by BigTako — APPROVED** [review]

File: src/app/ThemeProvider.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+'use client';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { ThemeProvider as Provider } from 'styled-components';
+import StyledComponentsRegistry from '../lib/registry';
+import theme from './styles/theme';
+
+const ThemeProvider = props => (
```
Comment (watchlar2000) [inline src/app/ThemeProvider.js:9]: I would spread the props just to follow the common code style

```
const ThemeProvider = ({children}) => (
  <StyledComponentsRegistry>
    <Provider theme={theme}>{children}</Provider>
  </StyledComponentsRegistry>
);
```

**Comment by alex-popov-tech** [comment]
closing this in favor of https://github.com/keenethics/mayak/pull/10 , as there we've added `tailwind` and required defaults

**Pull request closed without merging**